app/src/main/java/org/lineageos/twelve/database/dao/ResumptionPlaylistDao.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: 2024 The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.twelve.database.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Transaction
+import org.lineageos.twelve.database.entities.ResumptionPlaylistWithMediaItems
+
+@Dao
+@Suppress("FunctionName")
+interface ResumptionPlaylistDao {
+    /**
+     * Clear all resumption playlists.
+     */
+    @Query("DELETE FROM ResumptionPlaylist")
+    suspend fun _clearResumptionPlaylists()
+
+    /**
+     * Insert a resumption playlist.
+     */
+    @Query(
+        """
+            INSERT INTO ResumptionPlaylist (start_index, start_position_ms)
+            VALUES (:startIndex, :startPositionMs)
+        """
+    )
+    suspend fun _createResumptionPlaylist(startIndex: Int, startPositionMs: Long): Long
+
+    /**
+     * Add an item to a resumption playlist.
+     */
+    @Query(
+        """
+            INSERT OR REPLACE INTO ResumptionItem (playlist_index, resumption_playlist_id, media_id)
```

Comment:
`OR REPLACE` shouldn't be needed, the removal of a resumption_playlist will cascade delete all ResumptionItem